lib/module-index-client/src/lib.rs

```diff
@@ -232,11 +233,11 @@ impl ModuleIndexClient {
         Ok(upsert_response.json::<bool>().await?)
     }
 
-    pub async fn download_module(&self, module_id: Ulid) -> ModuleIndexClientResult<Vec<u8>> {
+    pub async fn download_module(&self, module_id: ModuleId) -> ModuleIndexClientResult<Vec<u8>> {
```

Comment:
This does seem to be a nice payoff--get some specific type safety around what kind of ID is expected!